project_from_quotation/models/sale_order.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _compute_project_ids(self):
+        """
+        Override related projects computation to
+        include projects containing tasks related to
+        order lines of this quotation
+        """
+        super()._compute_project_ids()
+        for order in self:
+            order.project_ids |= (
+                self.env["project.task"]
+                .search(
+                    [
+                        ("quotation_line_id", "in", order.order_line.ids),
+                    ]
+                )
+                .mapped("project_id")
+            )
+            order.project_count = len(order.project_ids)
+        return
+
+    @api.model
+    def _prepare_task_vals(
+        self,
+        project,
+        order_line,
+    ):
+        """
+        Prepare vals for task creation
+
+        Args:
+        project (project.project): Project
+        order_line (sale.order.line): Order line
+
+        Returns:
+            dict: Dictionary of task values
+        """
+        return {
+            "name": order_line.name,
+            "project_id": project.id,
+            "quotation_line_id": order_line.id,
+        }
+
+    @api.model
+    def _prepare_project_vals(self, sale_order):
+        """
+        Prepare vals for project creation
+
+        Args:
+        sale_order (sale.order): Sale Order
+
+        Returns:
+            dict: Dictionary of project values
+        """
+        return {
+            "name": sale_order.name,
+            "sale_order_id": sale_order.id,
+        }
```

Comment:
You need to add `"active": True`
Otherwise if archive project is used as a template created one will be archived too:
![image](https://github.com/OCA/sale-workflow/assets/2234999/b2d36dee-878c-435e-bab2-1eb82b603d7a)